lib/utils/patterns.ts

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2016-2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import _ = require('lodash');
+import Promise = require('bluebird');
+import form = require('resin-cli-form');
+import visuals = require('resin-cli-visuals');
+import ResinSdk = require('resin-sdk');
+import chalk from 'chalk';
+import validation = require('./validation');
+import messages = require('./messages');
+
+const resin = ResinSdk.fromSharedOptions();
+
+export function authenticate(options: {}) {
+	form.run([{
```

Comment:
Is it correct fine that we removed the `return`? Since, in [lib/actions/auth.coffee](https://github.com/resin-io/resin-cli/blob/810ca78215ee1dd71e0dd84506476e5af71da8be/lib/actions/auth.coffee#L92) it seems that we return the result of this function.

Reply:
Fixed, great catch! That would've broken manually entered logins, nice spot.